zerver/lib/streams.py

```diff
@@ -447,7 +447,23 @@ def check_for_exactly_one_stream_arg(stream_id: int | None, stream: str | None)
         raise IncompatibleParametersError(["stream_id", "stream"])
 
 
-def check_stream_access_for_delete_or_update(
+def channel_admin_has_content_access(
+    user_profile: UserProfile, stream: Stream, sub: Subscription | None = None
+) -> bool:
+    if not stream.invite_only or sub is not None:
+        return True
```

Comment:
I feel like we should not have code that checks `invite_only` and does not consider `is_guest`; it is only non-guests that have implicit access to public channels. (I know it's weird to have a guest who is a channel administrator, but for security code, we should be consistent about this sort of detail)